sandcastle/api.py

```diff
@@ -107,18 +110,15 @@ def __init__(
         self.k8s_namespace_name = k8s_namespace_name
 
         # regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?
-        self.cleaned_name = (
-            image_reference.replace("/", "-").replace(":", "-").replace(".", "-")
-        )
+        self.cleaned_name = clean_string(image_reference)
         if pod_name:
             self.pod_name = pod_name
```

Comment:
I would add there `get_timestamp_now()` always, so `self.pod_name` is unique even if it is defined by the user.

Reply:
ehm, why? if user specifies the name of the pod, and we produce a different name, how would the user be able to interact with the pod when the name he specified would not work?

Reply:
ok, not clearly expressed. I thought to avoid creating two pods with the same name. In case the older one is not deleted and the new one is going to be created and there will be conflicts.
We will see in a real situation if it happens.